File: 2265-count-nodes-equal-to-average-of-subtree/2265-count-nodes-equal-to-average-of-subtree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self,root,co):
    
        if not root:
            return 0,0,co

        l_s,l_c,co=self.dfs(root.left,co)
        r_s,r_c,co=self.dfs(root.right,co)

        sm=l_s+r_s+root.val
        c=l_c+r_c+1

        if sm//c == root.val:
            co+=1

        return sm,c,co

    
    def averageOfSubtree(self, root: TreeNode) -> int:
        co=0

        _,_,co=self.dfs(root,co)
        return co

File: 2265-count-nodes-equal-to-average-of-subtree/2265-count-nodes-equal-to-average-of-subtree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int count=0;

    int[] dfs(TreeNode node)
        {
            if(node==null)
            return new int[]{0,0};

            int[] lft=dfs(node.left);
            int[] rgh=dfs(node.right);

            int sum=node.val+lft[0]+rgh[0];
            int c=1+lft[1]+rgh[1];

            if(sum/c==node.val)
            {
                count++;
            }
            return new int[]{sum,c};
        }

    public int averageOfSubtree(TreeNode root) 
    {
        
        dfs(root);
        return count;
    }
}